Add Windows llvm debug build

mcbarton opened the pull request:
This PR should allow Clad to be tested against the head of release branch of llvm on Windows.

mcbarton committed:
Add Windows llvm debug build

vgvassilev reviewed (commented):
Looks like the bots are not picking up these changes, right?

mcbarton commented:
@vgvassilev They only build llvm on Windows if debug=true in the ci. If you look at the first commit where I forgot to add this condition, the ci is building llvm on Windows before I stopped the job.

vgvassilev commented:
> @vgvassilev They only build llvm on Windows if debug=true in the ci. If you look at the first commit where I forgot to add this condition, the ci is building llvm on Windows before I stopped the job.

Can we make a new configuration that builds llvm on windows, caches it and builds clad against it?

mcbarton commented:
@vgvassilev This PR is ready for review. If you look at the ci it uses the built cache to build Clad. If you merge, remember to clear the cache before you merge. I'll put in a PR at some point to reduce the size of the cache.

vgvassilev commented:
> @vgvassilev This PR is ready for review. If you look at the ci it uses the built cache to build Clad. If you merge, remember to clear the cache before you merge. I'll put in a PR at some point to reduce the size of the cache.

I think we will need a bit more to this. We need to compile with `clang-cl` to be able to pick up the windows exports. We should use something like `cmake -GNinja -DLLVM_ENABLE_PROJECTS=clang -DCMAKE_BUILD_TYPE=RelWithDebInfo -DLLVM_TARGETS_TO_BUILD=host -DLLVM_ENABLE_PLUGINS=On ..\llvm`

Then we need to compile clad with: `cmake -DCMAKE_BUILD_TYPE=Debug -DLLVM_DIR=c:\vv-llvm-dev\llvm-project\build -GNinja -DLLVM_BUILD_LLVM_DYLIB_VIS=On -DLLVM_LINK_LLVM_DYLIB=On -DCLANG_LINK_CLANG_DYLIB=On ..\clad`

mcbarton commented:
@vgvassilev I'll do this once you clear the cache for this PR. I'll also try and reduce the size of the cache at the same time.

vgvassilev commented:
> @vgvassilev I'll do this once you clear the cache for this PR. I'll also try and reduce the size of the cache at the same time.

Done!

mcbarton commented:
@vgvassilev I've made the changes you requested (using clang-cl to build llvm, and use certain options when building clad), but it fails with a linker error when building llvm. Any suggestions of a fix?

mcbarton committed:
Update ci.yml

mcbarton committed:
Update ci.yml

mcbarton committed:
Update ci.yml

vgvassilev commented:
Are there any news on this?

mcbarton committed:
Merge branch 'master' into Windows-llvm-build-ci

mcbarton committed:
Update CMakeLists.txt

mcbarton commented:
Now that the llvm this PR is building is llvm 20 (temporarily a fork of the llvm-project until the changes are upstream) this PR shall wait for https://github.com/vgvassilev/clad/pull/1267 to go in.

mcbarton committed:
Merge branch 'master' into Windows-llvm-build-ci

mcbarton committed:
Update ci.yml

mcbarton committed:
Merge branch 'master' into Windows-llvm-build-ci

mcbarton committed:
Update ci.yml

mcbarton committed:
Merge branch 'master' into Windows-llvm-build-ci

mcbarton committed:
Merge branch 'master' into Windows-llvm-build-ci

mcbarton committed:
Update ci.yml

mcbarton committed:
Update ci.yml

mcbarton committed:
Update ci.yml

mcbarton committed:
Merge branch 'master' into Windows-llvm-build-ci

mcbarton committed:
Merge branch 'master' into Windows-llvm-build-ci

mcbarton committed:
Merge branch 'master' into Windows-llvm-build-ci

mcbarton committed:
Merge branch 'master' into Windows-llvm-build-ci

mcbarton committed:
Merge branch 'master' into Windows-llvm-build-ci

mcbarton committed:
Merge branch 'master' into Windows-llvm-build-ci

mcbarton committed:
Merge branch 'master' into Windows-llvm-build-ci

mcbarton committed:
Turn off including examples

mcbarton committed:
Disable static analyzer in CI configuration

mcbarton committed:
Try previous llvm commit

mcbarton committed:
Update ci.yml

mcbarton committed:
Update ci.yml

mcbarton committed:
Change build command to target specific components

mcbarton committed:
Merge branch 'master' into Windows-llvm-build-ci

mcbarton committed:
Simplify cmake build command in CI workflow

Removed specific targets from the cmake build command to streamline the build process.
